### ISSUE TITLE
`vagrant rsync-auto` claims "There are no paths to watch" when using Docker provider, even though `vagrant rsync` finds them

### ISSUE THREAD

**Issue by motin** [hashicorp/vagrant#4067]
`vagrant up` and SSH-based provision works like a charm, but this is what happens after running `vagrant rsync-auto`:

```
$ vagrant rsync-auto web
==> web: New synced folders were added to the Vagrantfile since running
==> web: `vagrant reload`. If these new synced folders are backed by rsync,
==> web: they won't be automatically synced until a `vagrant reload` is run.
There are no paths to watch! This is either because you have no
synced folders using rsync, or any rsync synced folders you have
have specified `rsync_auto` to be false.

$     vagrant rsync
==> web: Rsyncing folder: /Users/motin/Dev/Projects/foo/ => /code
==> web:   - Exclude: [".vagrant/", ".git/"]
```

(here another error is encountered, but it is another issue: #4066)

Since vagrant recommends running reload, here is the continued output:

```
$ vagrant reload web
==> web: Stopping container...
A Docker command executed by Vagrant didn't complete successfully!
The command run along with the output from the command is shown
below.

Command: "docker" "stop" "-t" "1" "d68267b9b7a18a137dad8fb165b3d2ef484e3b22d09d514605d55fef6d400ff6"

Stderr: Error: Cannot stop container d68267b9b7a18a137dad8fb165b3d2ef484e3b22d09d514605d55fef6d400ff6: no such process
2014/06/19 07:49:39 Error: failed to stop one or more containers

Stdout:

$ vagrant reload web
==> web: Vagrant has noticed that the synced folder definitions have changed.
==> web: With Docker, these synced folder changes won't take effect until you
==> web: destroy the container and recreate it.
==> web: Starting container...
==> web: Waiting for machine to boot. This may take a few minutes...
==> web: Machine booted and ready!
==> web: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> web: to force provisioning. Provisioners marked to run always will still run.

$ docker rm -f d4ef13cfc533 # destroying the container
d4ef13cfc533

$ vagrant up web
Bringing machine 'web' up with 'docker' provider...
==> web: Docker host is required. One will be created if necessary...
    web: Docker host VM is already ready.
==> web: Syncing folders to the host VM...
    web: Rsyncing folder: /Users/motin/Dev/Projects/foo/ => /var/lib/docker/docker_1403164320_27210
    web:   - Exclude: [".vagrant/", ".git/"]
    web: Mounting shared folders...
    web: /var/lib/docker/docker_1403164320_37155 => /Users/motin/Dev/Projects/bar/virtual-machines/vagrant/cms/build/cms-develop-virtualbox
==> web: Warning: When using a remote Docker host, forwarded ports will NOT be
==> web: immediately available on your machine. They will still be forwarded on
==> web: the remote machine, however, so if you have a way to access the remote
==> web: machine, then you should be able to access those ports there. This is
==> web: not an error, it is only an informational message.
==> web: Creating the container...
    web:   Name: cms-develop-virtualbox_web_1403164346
    web:  Image: myteam/cms:develop-clean-db
    web:    Cmd: /bin/bash /vagrant/web-start.sh
    web: Volume: /var/lib/docker/docker_1403164320_27210:/code
    web: Volume: /var/lib/docker/docker_1403164320_37155:/vagrant
    web:   Port: 2222:22
    web:   Port: 11111:5000
    web:   Port: 2202:2202
    web:
    web: Container created: 743e3a1cb8aacf0e
==> web: Starting container...
==> web: Waiting for machine to boot. This may take a few minutes...
==> web: Machine booted and ready!
==> web: Running provisioner: shell...
    web: Running: /var/folders/x1/x3gztr1d5z798jf6t_xg605m0000gn/T/vagrant-shell20140619-28749-4fx24d.sh
==> web: + cd /vagrant
==> web: + '[' '!' -d provisioning ']'
==> web: + cd provisioning
==> web: + apt-get update
==> web: Ign http://archive.ubuntu.com quantal InRelease
# etc - successful provisioning...

$ vagrant rsync-auto web
==> web: New synced folders were added to the Vagrantfile since running
==> web: `vagrant reload`. If these new synced folders are backed by rsync,
==> web: they won't be automatically synced until a `vagrant reload` is run.
There are no paths to watch! This is either because you have no
synced folders using rsync, or any rsync synced folders you have
have specified `rsync_auto` to be false.
```

Vagrantfile:

```
# -*- mode: ruby -*-
# vi: set ft=ruby :

# Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
VAGRANTFILE_API_VERSION = "2"
ENV['VAGRANT_DEFAULT_PROVIDER'] = 'docker'

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|

    config.ssh.private_key_path = 'insecure_key'

    # run general provisioning script
    config.vm.provision "shell", :path => "provision.sh"

    # containers should always boot within 30 seconds, timeout otherwise since it would otherwise take 300 seconds (default) to timeout
    config.vm.boot_timeout = 30

    config.vm.provider :docker do |provider|
        # use a custom host vm vagrant file in order to gain ssh access and port forwarding
        provider.vagrant_vagrantfile = "./host-vm/Vagrantfile"
        # make sure that linux users also spins up the host vm for consistency's sake (remove this if it works just as well without it)
        provider.force_host_vm
    end

    config.vm.define "web" do |web|

        #
        web.vm.synced_folder "../../../../../cms/", "/code",
          type: "rsync",
          rsync__exclude: [
            ".git/"
          ],
          rsync__auto: true
        #

        web.vm.provider :docker do |provider|
            provider.image = "myteam/cms:develop-clean-db"
            provider.ports = ["11111:5000", "2202:2202"]
            provider.cmd = ["/bin/bash", "/vagrant/web-start.sh"]
            provider.has_ssh = true
        end

        # since the image doesn't have a vagrant user
        web.ssh.username = 'root'

        # docker-container-ssh.sh starts the ssh server on port 2202 to avoid conflicts and confusion with ports 22 and 2222
        web.ssh.port = 2202

        # cms web-instance provisioning is not done through vagrant's ssh provisioning mechanism since
        # the docker container is expected to be provisioned before (or at the latest in the process of) being started
        # web-provision.sh is instead run by web-start.sh before starting the container
        # web.vm.provision "shell", :path => "web-provision.sh"

    end

    config.vm.define "db" do |db|

        db.vm.provider :docker do |provider|
            provider.image = "mariadb/develop-cms"
            provider.ports = ["13306:3306"]
            provider.cmd = ["/usr/bin/start_mariadb.sh", "dbpassword"]
            provider.volumes = ["/var/lib/docker/vfs/dir/4a7d01cf2ea228e4bca4a1af3497923a761e703cc360f390d6d0111952afd72a:/opt/mysql"]
            #provider.has_ssh = true
        end

    end

end
```

host-vm/Vagrantfile:

```
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|

    # base box for virtualbox/vmware
    config.vm.box = "ubuntu/trusty64"

    # using the Docker provisioner will automatically install Docker
    config.vm.provision "docker"

    # Enable Docker daemon listen on TCP :4243
    config.vm.provision :shell, inline: <<-EOF
        service ssh restart

        echo "Enable Docker daemon listen on TCP :4243" >&2
        echo 'DOCKER_OPTS="-H tcp://0.0.0.0:4243 -H unix:///var/run/docker.sock"' >> /etc/default/docker

        service docker restart
    EOF
    config.vm.provision "shell", inline: "ps aux | grep 'sshd:' | awk '{print $2}' | xargs kill"

    # configure docker port so that we can set DOCKER_HOST to tcp://localhost:4243 locally on osx and access docker commands
    config.vm.network :forwarded_port, guest: 4243, host: 4243

    # port forward necessary for http://localhost:$WEB_PORT to work on local machine
    config.vm.network :forwarded_port, guest: 11111, host: 11111

    # port forward necessary to be able to SSH directly into docker web container (for troubleshooting)
    config.vm.network :forwarded_port, guest: 2202, host: 2202

    # port forward necessary to be able to connect to the mysql server against 127.0.0.1:$DB_PORT locally
    config.vm.network :forwarded_port, guest: 13306, host: 13306

end
```
